service/population.go

```diff
@@ -55,3 +56,23 @@ func (p *populationService) GetPopulationStatistics() ([]model.PopulationRespons
 	}
 	return out, nil
 }
+
+func (p *populationService) GetAllCandidateInfo() ([]model.PopulationDatabaseRow, error) {
+	out := []model.PopulationDatabaseRow{}
+	districts, err := p.repository.QueryAllDistrict()
+	if err != nil {
+		logrus.Error(err)
+	}
+	for _, s := range districts {
+		districtId := s.DistrictID
+		res, err := p.repository.QueryCandidateByDistrict(districtId)
+		if err != nil {
+			logrus.Error(err)
+			continue
+		}
+		for _, c := range res {
+			out = append(out, c)
+		}
+	}
+	return out, nil
```

Comment:
I don't think we should return an incomplete candidate list in case of errors apart from the candidate not found.

Reply:
Should check the error type and handle it correctly.

Reply:
fixed 🔧